### PR TITLE
fix(user_tags): Fix clicking affected users with special characters

### PR DESCRIPTION
**Pull request by wmak**
- This is to fix an issue when clicking an affected user in an event tag
  with a space in it, currently the query is missing quotes, so we
  interpret the part after a space as another term.
    - eg. user.id:123 456
- As part of this also fixing it when the tag has a quote in it as well
    - eg. user.id:123"456
- [Jira Ticket](https://getsentry.atlassian.net/browse/ISSUE-605)

### CODE REVIEW

**Review by dashed — APPROVED** [review]
Looks good to me 👍

File: tests/sentry/search/test_utils.py [hunk 1]
```diff
@@ -514,3 +515,17 @@ def test(self):
             environment = self.create_environment()
             result = get_latest_release([self.project], [environment])
             assert result == new.version
+
+
+class ConvertUserTagTest(TestCase):
```
Comment (wedamija) [inline tests/sentry/search/test_utils.py]: Thanks for adding direct tests, I guess I tested this via serializer when I added it

**Review by wedamija — APPROVED** [review]

**Comment by wedamija** [comment]
Actually, isn't the point of this ticket that we need to start actually using `query` on the FE?

**Comment by wmak** [comment]
@wedamija Might be that I'm misunderstanding how it works-- but I think the FE already uses `query`?

File: tests/sentry/search/test_utils.py [hunk 2]
```diff
@@ -514,3 +515,17 @@ def test(self):
...
+    def test_simple_user_tag(self):
+        assert convert_user_tag_to_query("user", "id:123456") == 'user.id:"123456"'
+
+    def test_user_tag_with_quote(self):
+        assert convert_user_tag_to_query("user", 'id:123"456') == 'user.id:"123\\"456"'
+
+    def test_user_tag_with_space(self):
+        assert convert_user_tag_to_query("user", "id:123 456") == 'user.id:"123 456"'
```
Comment (markstory) [inline tests/sentry/search/test_utils.py]: black is likely going to get salty about the single quoted strings.

Comment (wmak) [reply tests/sentry/search/test_utils.py]: So Black flipped me to the single quoted strings cause originally I had
`"user.id:\"123 456\""`

Comment (markstory) [reply tests/sentry/search/test_utils.py]: lololol.

**Comment by wedamija** [comment]
> @wedamija Might be that I'm misunderstanding how it works-- but I think the FE already uses `query`?

I'm not sure where we got with that - if FE made the move to using `query` then we're all good.

**Comment by wmak** [comment]
Just to be safe. Did some extra searching, looks like it was flipped from this PR: https://github.com/getsentry/sentry/pull/12204

**Pull request merged**